front/app/components/ParticipationCTABars/VotingCTABar/CTAButton.tsx

```diff
@@ -22,35 +23,60 @@ import useLocalize from 'hooks/useLocalize';
 import JSConfetti from 'js-confetti';
 import { scrollToElement } from 'utils/scroll';
 import { getDisabledExplanation } from './utils';
+import clHistory from 'utils/cl-router/history';
 
 // typings
 import { IProjectData } from 'api/projects/types';
 import { IPhaseData } from 'api/phases/types';
 
 const confetti = new JSConfetti();
 
+const StyledButton = styled(Button)`
+  &.pulse {
+    animation-name: pulse;
+    animation-duration: 1.3s;
+    animation-timing-function: ease;
+    animation-delay: 0.3s;
+    animation-iteration-count: infinite;
+  }
+
+  /* border-radius */
+  -webkit-border-radius: 3px;
+  -moz-border-radius: 3px;
+  border-radius: 3px;
+
+  @keyframes pulse {
+    0% {
+      box-shadow: 0 0 0 0 rgba(255, 255, 255, 1);
+    }
+    100% {
+      box-shadow: 0 0 0 6px rgba(255, 255, 255, 0);
+    }
+  }
+`;
 interface Props {
   participationContext: IProjectData | IPhaseData;
+  projectId?: string;
 }
 
-const CTAButton = ({ participationContext }: Props) => {
-  const [processing, setProcessing] = useState(false);
+const CTAButton = ({ participationContext, projectId }: Props) => {
+  const theme = useTheme();
+  const { formatMessage } = useIntl();
+  const localize = useLocalize();
+  const { data: appConfig } = useAppConfiguration();
+  const { data: project } = useProjectById(projectId);
 
   const basketId = participationContext.relationships.user_basket?.data?.id;
-
   const { data: basket } = useBasket(basketId);
   const { mutate: updateBasket } = useUpdateBasket();
   const { numberOfVotesCast, processing: votingProcessing } = useVoting();
-  const { data: appConfig } = useAppConfiguration();
-  const theme = useTheme();
-  const { formatMessage } = useIntl();
-  const localize = useLocalize();
+
+  const [processing, setProcessing] = useState(false);
+  const currency = appConfig?.data.attributes.settings.core.currency;
```

Comment:
I find it helps with readability for me when it's used later on (even if it's only in one place), since it's a simple "currency" rather than "appConfig?.data.attributes.settings.core.currency" :thinking:

Do you know if we have a convention in the team here? :)

Reply:
No convention for this I'm aware of. I still find the long value fine to read if it's only used 1x, besides you have the immediate value and don't need to look up where it comes from. But my preferences have changed and continue to change, so I can empathise.